packages/dialect-react-ui/components/Chat/screens/Main/index.tsx

```diff
@@ -1,77 +1,92 @@
-import React, { useState } from 'react';
 import clsx from 'clsx';
 import { useDialect } from '@dialectlabs/react';
-import { useTheme } from '../../../common/ThemeProvider';
-import IconButton from '../../../IconButton';
-import CreateThread from './pages/CreateThreadPage/CreateThread';
-import ThreadPage from './pages/ThreadPage/';
+import { useTheme } from '../../../common/providers/DialectThemeProvider';
 import ThreadsList from './ThreadsList';
+import { Header } from '../../../Header';
+import CreateThread from '../CreateThreadPage/CreateThread';
+import ThreadPage from '../ThreadPage';
+import { useChatInternal } from '../../provider';
+import { Route, Router, useRoute } from '../../../common/providers/Router';
+import { MainRouteName, RouteName, ThreadRouteName } from '../../constants';
+import { useDialectUiId } from '../../../common/providers/DialectUiManagementProvider';
 
-interface MainProps {
-  inbox?: boolean;
-  onModalClose?: () => void;
-}
-
-const Main = ({ inbox, onModalClose }: MainProps) => {
+const Main = () => {
+  const { navigate, current } = useRoute();
   const { dialectAddress, dialects, setDialectAddress } = useDialect();
+  const { type, onChatClose, onChatOpen, id } = useChatInternal();
+  const { ui } = useDialectUiId(id);
+  const inbox = type === 'inbox';
 
   const { icons } = useTheme();
 
-  const [newThreadOpen, setNewThreadOpen] = useState(false);
-
   return (
     <div className="dt-h-full dt-flex dt-flex-1 dt-justify-between dt-w-full">
       <div
         className={clsx(
           'dt-flex dt-flex-1 dt-flex-col dt-border-neutral-600 dt-overflow-hidden dt-w-full',
           {
             'md:dt-max-w-xs md:dt-border-r md:dt-flex': inbox,
-            'dt-hidden': dialectAddress || newThreadOpen,
+            'dt-hidden':
+              dialectAddress ||
+              current?.sub?.name === MainRouteName.CreateThread, // TODO: ideally we should control this with routes and this causes a visual bug with route swaps
```

Comment:
just wonder, what kind of visual bug?

Reply:
Router provider is setting the child `activeRoute` in `useEffect`, so route on lower levels happens slower than this class update. That causes a quick appearance of the menu, while we are still on the `CreateThread` route.

Reply:
have you tried useLayoutEffect?

Reply:
useLayoutEffect will cause problems (afaik) with SSR projects, at least I remember a bunch of warnings at the time.

But I'll give it a shot, thanks for the suggestion

Reply:
oh I see, then should be smth like this https://github.com/streamich/react-use/blob/master/src/useIsomorphicLayoutEffect.ts#L4
to support ssr projects

Reply:
Yes, good point! 

Btw, useLayoutEffect is working! Nextjs isn't sending any warnings and actually it makes sense, I believe, since we don't SSR render this page on the backend

Reply:
cool, but anyway I think it should be isomorphic, since some of our partners could use ssr ;)